### DISCUSSION TITLE
Diagonal Born-Oppenheimer Correction

### DISCUSSION THREAD

**Discussion by Physfock** [pyscf/pyscf#2641] [Q&A]
Hey there! I was wondering if anyone could help me with a little problem I'm having. I'm trying to calculate the diagonal Born-Oppenheimer correction for the H2 molecule with pyscf using the built-in functions. Would anyone be so kind as to show me how to do this at the CCSD or CISD level? Thanks in advance! 😊

**Comment by fishjojo** [comment]
You will need to compute the nonadiabatic couplings. 
There is an implementation for the MS-PDFT method in https://github.com/pyscf/pyscf-forge/blob/master/pyscf/nac/mspdft.py, which you may find useful.
But for CCSD and CISD, we don't have them at the moment.

**Comment by matthew-hennefarth** [comment]
There is also an implementation for NACs in this repository for regular SA-CASSCF. Example [`examples/nac/01-sacasscf_nac.py`](https://github.com/pyscf/pyscf/blob/master/examples/nac/01-sacasscf_nac.py).

Also, this issue is related to #735; though I think with different outcomes desired.

**Comment by ndattani** [comment]
Another person recently asked [here](https://mattermodeling.stackexchange.com/q/14178/5) about the potential for DBOCs to be calculated in PySCF. For a (very small) list of software that can do them, see [here](https://mattermodeling.stackexchange.com/q/9204/5).
